src/client/kwil.ts

```diff
@@ -193,21 +153,28 @@ export abstract class Kwil<T extends EnvironmentType> extends Client {
   ): Promise<GenericResponse<TxReceipt>> {
     const name = !actionBody.name && actionBody.action ? actionBody.action : actionBody.name;
 
-    let tx = ActionBuilderImpl.of<T>(this)
-      .dbid(actionBody.dbid)
-      .name(name)
-      .description(actionBody.description || '')
-      .publicKey(kwilSigner.identifier)
-      .chainId(this.chainId)
-      .signer(kwilSigner.signer, kwilSigner.signatureType);
+    let tx = Action.createTx(this, {
+      dbid: actionBody.dbid,
+      actionName: name.toLowerCase(),
+      description: actionBody.description || '',
+      identifier: kwilSigner.identifier,
+      chainId: this.chainId,
+      signer: kwilSigner.signer,
+      signatureType: kwilSigner.signatureType,
+      nonce: actionBody.nonce,
+    });
 
     if (actionBody.inputs) {
-      const inputs = resolveActionInputs(actionBody.inputs);
-      tx = tx.concat(inputs);
-    }
-
-    if (actionBody.nonce) {
-      tx = tx.nonce(actionBody.nonce);
+      const inputs = (actionBody.inputs as ActionInput[]).every(
```

Comment:
Type casting: ![9bqtu0](https://github.com/user-attachments/assets/e5629584-f205-4cfc-82d9-38320201accc)

Reply:
Seriously though, try to write without `as`.